bluetooth/src/iosMain/kotlin/scanner/Scanner.kt

```diff
@@ -193,4 +200,28 @@ actual class Scanner internal constructor(
             )
         }
     }
+
+    class ScanSettings private constructor(val scanOptions: Map<Any?, *>) {
+        data class Builder(
+            // https://developer.apple.com/documentation/corebluetooth/cbcentralmanagerscanoptionallowduplicateskey
+            var allowDuplicateKeys: Boolean = true,
+            // https://developer.apple.com/documentation/corebluetooth/cbcentralmanagerscanoptionsolicitedserviceuuidskey
+            var solicitedServiceUUIDsKey: List<UUID>? = null,
+        ) {
+            fun allowDuplicateKeys(allow: Boolean) =
+                apply { allowDuplicateKeys = allow }
+
+            fun solicitedServiceUUIDsKey(keys: List<UUID>) =
+                apply { solicitedServiceUUIDsKey = keys }
+
+            fun build(): ScanSettings {
+                val result: MutableMap<String, Any> =
```

Comment:
still no need for `Any` or a `Map` here I'd say, can just use fields right?

Reply:
Hi @thoutbeckers, scanForPeripheralsWithServices takes:

public open external fun scanForPeripheralsWithServices(
    serviceUUIDs: List<*>?,
    options: Map<Any?, *>?
): Unit

So it made sense to me to do:

class ScanSettings private constructor(val scanOptions: Map<Any?, *>). 

Could you please elaborate with what you mean just fields?